### PR TITLE
Macie integration tests

### PR DESCRIPTION
**Pull request by nico-stefani**
|Related issue|
|-------------|
|      #3340      |

## Description

This PR adds tests for AWS Macie integration

#### Tier 0
- Default configuration tests
- `remove_from_bucket` parameter tests
- `only_logs_after` parameter tests
- `path` parameter tests
- `discard_regex` parameter tests

#### Tier 1
- `only_logs_after` parameter tests
---

## Testing performed

| Tester             | Test path | Jenkins | Local  | OS | Commit | Notes                |
|--------------------|-----------|---------|--------|-----|--------|----------------------|
| @nico-stefani  (Developer)  | `tests/integration/test_aws` | ⚫⚫⚫ | :green_circle: :red_circle:  | Ubuntu 22.04    |  [^1]    | Nothing to highlight |
| @reviewer (Reviewer)   | | ⚫⚫⚫ |  |   |      | Nothing to highlight |

### Tier 0 results
```python
root@ubuntu-jammy:/home/vagrant/qa/tests/integration/test_aws# pytest -k macie --tier 0
================================================================================================ test session starts ================================================================================================
platform linux -- Python 3.10.6, pytest-7.1.2, pluggy-1.0.0
rootdir: /home/vagrant/qa/tests/integration, configfile: pytest.ini
plugins: metadata-2.0.2, html-3.1.1, testinfra-5.0.0
collected 24 items / 16 deselected / 8 selected

test_basic.py .                                                                                                                                                                                               [ 12%]
test_discard_regex.py .                                                                                                                                                                                       [ 25%]
test_only_logs_after.py ..                                                                                                                                                                                    [ 50%]
test_path.py ...                                                                                                                                                                                              [ 87%]
test_remove_from_bucket.py .                                                                                                                                                                                  [100%]

================================================================================================= warnings summary ==================================================================================================
../../../../../../usr/local/lib/python3.10/dist-packages/_pytest/cacheprovider.py:433
  /usr/local/lib/python3.10/dist-packages/_pytest/cacheprovider.py:433: PytestCacheWarning: cache could not write path /home/vagrant/qa/tests/integration/.pytest_cache/v/cache/nodeids
    config.cache.set("cache/nodeids", sorted(self.cached_nodeids))

../../../../../../usr/local/lib/python3.10/dist-packages/_pytest/stepwise.py:52
  /usr/local/lib/python3.10/dist-packages/_pytest/stepwise.py:52: PytestCacheWarning: cache could not write path /home/vagrant/qa/tests/integration/.pytest_cache/v/cache/stepwise
    session.config.cache.set(STEPWISE_CACHE_DIR, [])

-- Docs: https://docs.pytest.org/en/stable/how-to/capture-warnings.html
============================================================================= 8 passed, 16 deselected, 2 warnings in 164.31s (0:02:44) ==============================================================================
```

### Tier 1 results

```python
root@ubuntu-jammy:/home/vagrant/qa/tests/integration/test_aws# pytest -k macie --tier 1
============================= test session starts ==============================
platform linux -- Python 3.10.6, pytest-7.1.2, pluggy-1.0.0
rootdir: /home/vagrant/qa/tests/integration, configfile: pytest.ini
plugins: metadata-2.0.2, html-3.1.1, testinfra-5.0.0
collected 24 items / 23 deselected / 1 selected

test_only_logs_after.py F                                                [100%]
------------------------------ Captured log call -------------------------------
DEBUG    wazuh_testing:cli_utils.py:23 Calling AWS module with: '[PosixPath('/var/ossec/wodles/aws/aws-s3'), '--bucket', 'wazuh-macie-integration-tests', '--type', 'custom', '--regions', 'us-east-1', '--aws_profile', 'qa', '--debug', '2']'
DEBUG    wazuh_testing:cli_utils.py:52 DEBUG: +++ Debug mode on - Level: 2
DEBUG    wazuh_testing:cli_utils.py:52 DEBUG: Generating default configuration for retries: mode standard - max_attempts 10
DEBUG    wazuh_testing:cli_utils.py:52 DEBUG: +++ Table does not exist; create
DEBUG    wazuh_testing:cli_utils.py:52 DEBUG: +++ Marker: 2023/01/16
DEBUG    wazuh_testing:cli_utils.py:52 DEBUG: ++ Skipping file with another prefix: test_prefix/2022/11/21/firehose_macie-1-20221121T1513Z_a301734c-4d08-4df8-b2de-4298f2539b08.json
DEBUG    wazuh_testing:cli_utils.py:52 DEBUG: +++ DB Maintenance
DEBUG    wazuh_testing:cli_utils.py:23 Calling AWS module with: '[PosixPath('/var/ossec/wodles/aws/aws-s3'), '--bucket', 'wazuh-macie-integration-tests', '--type', 'custom', '--regions', 'us-east-1', '--aws_profile', 'qa', '--debug', '2', '--only_logs_after', '2022-NOV-20']'
DEBUG    wazuh_testing:cli_utils.py:52 DEBUG: +++ Debug mode on - Level: 2
DEBUG    wazuh_testing:cli_utils.py:52 DEBUG: Generating default configuration for retries: mode standard - max_attempts 10
DEBUG    wazuh_testing:cli_utils.py:52 DEBUG: +++ Marker: 2022/11/20
DEBUG    wazuh_testing:cli_utils.py:52 DEBUG: ++ Found new log: 2022/11/21/firehose_macie-1-20221121T1513Z_a301734c-4d08-4df8-b2de-4298f2539b08.json
DEBUG    wazuh_testing:cli_utils.py:52 DEBUG: ++ Found new log: 2022/11/23/firehose_macie-1-20221123T1513Z_a301734c-4d08-4df8-b2de-4298f2539b08.json
DEBUG    wazuh_testing:cli_utils.py:52 DEBUG: ++ Found new log: 2022/11/26/firehose_macie-1-20221126T1513Z_a301734c-4d08-4df8-b2de-4298f2539b08.json
DEBUG    wazuh_testing:cli_utils.py:52 DEBUG: ++ Skipping file with another prefix: test_prefix/2022/11/21/firehose_macie-1-20221121T1513Z_a301734c-4d08-4df8-b2de-4298f2539b08.json
DEBUG    wazuh_testing:cli_utils.py:52 DEBUG: +++ DB Maintenance
DEBUG    wazuh_testing:cli_utils.py:23 Calling AWS module with: '[PosixPath('/var/ossec/wodles/aws/aws-s3'), '--bucket', 'wazuh-macie-integration-tests', '--type', 'custom', '--regions', 'us-east-1', '--aws_profile', 'qa', '--debug', '2', '--only_logs_after', '2022-NOV-20']'
DEBUG    wazuh_testing:cli_utils.py:52 DEBUG: +++ Debug mode on - Level: 2
DEBUG    wazuh_testing:cli_utils.py:52 DEBUG: Generating default configuration for retries: mode standard - max_attempts 10
DEBUG    wazuh_testing:cli_utils.py:52 DEBUG: +++ Marker: 2022/11/20
DEBUG    wazuh_testing:cli_utils.py:52 DEBUG: ++ Skipping previously processed file: 2022/11/21/firehose_macie-1-20221121T1513Z_a301734c-4d08-4df8-b2de-4298f2539b08.json
DEBUG    wazuh_testing:cli_utils.py:52 DEBUG: ++ Skipping previously processed file: 2022/11/23/firehose_macie-1-20221123T1513Z_a301734c-4d08-4df8-b2de-4298f2539b08.json
DEBUG    wazuh_testing:cli_utils.py:52 DEBUG: ++ Skipping previously processed file: 2022/11/26/firehose_macie-1-20221126T1513Z_a301734c-4d08-4df8-b2de-4298f2539b08.json
DEBUG    wazuh_testing:cli_utils.py:52 DEBUG: ++ Skipping file with another prefix: test_prefix/2022/11/21/firehose_macie-1-20221121T1513Z_a301734c-4d08-4df8-b2de-4298f2539b08.json
DEBUG    wazuh_testing:cli_utils.py:52 DEBUG: +++ DB Maintenance
ERROR    wazuh_testing:cli_utils.py:62 Some logs may were processed
ERROR    wazuh_testing:cli_utils.py:63 Results found: 3
ERROR    wazuh_testing:cli_utils.py:64 Results expected: 1
=========================== short test summary info ============================
FAILED test_only_logs_after.py::test_multiple_calls[macie_only_logs_after_multiple_calls]
================ 1 failed, 23 deselected, 3 warnings in 25.28s =================
```
[^1]: Because the module is showing the message `Skipping file with another prefix: ...` multiple times, the *tier 1* test is falling. This should be fixed in his respective issue.

### CODE REVIEW

File: deps/wazuh_testing/wazuh_testing/modules/aws/data_generator.py [hunk 1]
```diff
@@ -94,22 +103,258 @@ def get_data_sample(self) -> dict:
                     'sharedEventID': str(uuid4())
                 }
             ]
-        }
+        })
+
+
+class VPCDataGenerator(DataGenerator):
+    BASE_PATH = f'{cons.AWS_LOGS}/{cons.RANDOM_ACCOUNT_ID}/{cons.VPC_FLOW_LOGS}/{cons.US_EAST_1_REGION}/'
```
Comment (vicferpoy) [inline deps/wazuh_testing/wazuh_testing/modules/aws/data_generator.py:110]: F-strings must use double quotes as long as it is possible.

File: deps/wazuh_testing/wazuh_testing/modules/aws/data_generator.py [hunk 2]
```diff
@@ -94,22 +103,258 @@ def get_data_sample(self) -> dict:
...
+        <prefix>/AWSLogs/<suffix>/<organization_id>/<account_id>/vpcflowlogs/<region>/<year>/<month>/<day>
+        """
+        now = datetime.now()
+        path = f'{self.BASE_PATH}{now.strftime(cons.PATH_DATE_FORMAT)}/'
+        name = (
+            f'{self.BASE_FILE_NAME}{cons.FLOW_LOG_ID}_{now.strftime(cons.FILENAME_DATE_FORMAT)}_{abs(hash(now))}'
+            f'{cons.LOG_EXT}'
+        )
+
+        return f'{path}{name}'
```
Comment (vicferpoy) [inline deps/wazuh_testing/wazuh_testing/modules/aws/data_generator.py:124]: Use `os.path` instead.

File: deps/wazuh_testing/wazuh_testing/modules/aws/data_generator.py [hunk 3]
```diff
@@ -94,22 +103,258 @@ def get_data_sample(self) -> dict:
...
 
     Args:
         bucket_type (str): Bucket type to match the data generator.
+        bucket_name (str): Bucket name to match in case of custom types.
 
     Returns:
         DataGenerator: Data generator for the given bucket.
     """
+    if bucket_type == cons.CUSTOM_TYPE:
+        bucket_type = bucket_name.split('-')[1]
```
Comment (vicferpoy) [inline deps/wazuh_testing/wazuh_testing/modules/aws/data_generator.py:358]: What would happen if the `bucket_name` has more than one `-`? Consider using `split("-", 1)` to fix that, unless you   assume there are several `-` and only want the second element between them.

File: deps/wazuh_testing/wazuh_testing/tools/utils.py [hunk 4]
```diff
@@ -115,6 +115,15 @@ def get_random_ip():
     return fr"{randint(0,255)}.{randint(0,255)}.{randint(0,255)}.{randint(0,255)}"
 
 
+def get_random_port():
```
Comment (vicferpoy) [inline deps/wazuh_testing/wazuh_testing/tools/utils.py:118]:
```suggestion
def get_random_port() -> str:
```

File: deps/wazuh_testing/wazuh_testing/tools/utils.py [hunk 5]
```diff
@@ -162,12 +171,13 @@ def get_host_name():
 def validate_interval_format(interval):
     """Validate that the interval passed has the format in which the last digit is a letter from those passed and
        the other characters are between 0-9"""
-    if interval=='':
+    if interval == '':
         return False
-    if interval[-1] not in ['s','m', 'h','d','w','y'] or not isinstance(int(interval[0:-1]), numbers.Number):
+    if interval[-1] not in ['s', 'm', 'h', 'd', 'w', 'y'] or not isinstance(int(interval[0:-1]), numbers.Number):
```
Comment (vicferpoy) [inline deps/wazuh_testing/wazuh_testing/tools/utils.py:176]: I don't understand the second condition. Could you elaborate what is your intention?

**Review by vicferpoy — CHANGES_REQUESTED** [review]
Please be sure to apply some of the suggestions to every line that applies, not only the ones highlighted. Also, review all the missing typings and docstrings.

File: deps/wazuh_testing/wazuh_testing/modules/aws/data_generator.py (hunk 1 above)
Comment (nico-stefani) [reply deps/wazuh_testing/wazuh_testing/modules/aws/data_generator.py:110]: I need to cherry-pick some functionality included in https://github.com/wazuh/wazuh-qa/pull/3699 . This correction has already been fixed in 258068a66a1a5bf055be4574674df62e8841f9ac and will be included in [3333](https://github.com/wazuh/wazuh-qa/tree/3333-aws-integration-tests)

File: deps/wazuh_testing/wazuh_testing/modules/aws/data_generator.py (hunk 2 above)
Comment (nico-stefani) [reply deps/wazuh_testing/wazuh_testing/modules/aws/data_generator.py:124]: The same as I said before.

File: deps/wazuh_testing/wazuh_testing/modules/aws/data_generator.py (hunk 3 above)
Comment (nico-stefani) [reply deps/wazuh_testing/wazuh_testing/modules/aws/data_generator.py:358]: Yes, I'm assuming the second case.

File: deps/wazuh_testing/wazuh_testing/tools/utils.py (hunk 5 above)
Comment (nico-stefani) [reply deps/wazuh_testing/wazuh_testing/tools/utils.py:176]: This line has already been there. I just fixed some linter issues.

**Review by vicferpoy — APPROVED** [review]
LGTM

File: deps/wazuh_testing/wazuh_testing/modules/aws/data_generator.py [hunk 6]
```diff
@@ -497,6 +497,113 @@ def get_data_sample(self) -> str:
         )
 
 
+class MacieDataGenerator(DataGenerator):
+    BASE_PATH = ''
+    BASE_FILE_NAME = 'firehose_macie-1-'
+
+    def get_filename(self) -> str:
+        """Return the filename in the Macie format
```
Comment (davidjiglesias) [inline deps/wazuh_testing/wazuh_testing/modules/aws/data_generator.py:505]:
```suggestion
        """Return the filename in the Macie format.
```

**Review by davidjiglesias — CHANGES_REQUESTED** [review]

**Review by davidjiglesias — APPROVED** [review]

**Pull request merged**